mcs/class/test-helpers/SocketResponder.cs

```diff
@@ -91,8 +91,11 @@ private void Listen ()
 					listenSocket = tcpListener.AcceptSocket ();
 					listenSocket.Send (requestHandler (listenSocket));
 					try {
-						listenSocket.Shutdown (SocketShutdown.Receive);
+						// On Windows a Receive() is needed here before Shutdown() to consume the data some tests send.
```

Comment:
I have never seen it on other platforms. It does happen in 1 out of 100 or so on .NET when I run one of the tests that constantly fails on Mono on Windows. But I haven't tested this on any other platforms. Let me see if I can do that...